org.osgi.util.converter/src/org/osgi/util/converter/ConvertingImpl.java

```diff
@@ -216,6 +217,8 @@ else if (rt instanceof ParameterizedType) {
 			return convertToCollectionType(converter);
 		} else if (isMapType(targetAsClass, targetAsJavaBean, targetAsDTO)) {
 			return (T) convertToMapType(converter);
+		} else if (Optional.class.isAssignableFrom(targetAsClass)) {
```

Comment:
Optional classes are final, and this method misses other types like OptionalInt - the if check should cover these.

Reply:
you mean i should use `Optional.class.equals(targetAsClass)`